src/applications/bmqstoragetool/m_bmqstoragetool_journalfileprocessor.t.cpp

```diff
@@ -0,0 +1,1129 @@
+        ASSERT(journalFileIt.isReverseMode());
+        ASSERT_EQ(m_bmqstoragetool::moveToLowerBound(&journalFileIt, ts2), 1);
+        ResultChecker::check(journalFileIt, ts2);
+    }
+
+    {
+        // Timestamp more than last record in the file
+        const bsls::Types::Uint64 ts = numRecords * 2 *
+                                       journalFile.timestampIncrement();
+        mqbs::JournalFileIterator journalFileIt(
+            &journalFile.mappedFileDescriptor(),
+            journalFile.fileHeader(),
+            false);
+        // Move the iterator to the beginning of the file
+        ASSERT_EQ(journalFileIt.nextRecord(), 1);
+        ASSERT_EQ(m_bmqstoragetool::moveToLowerBound(&journalFileIt, ts), 0);
+        ASSERT_EQ(journalFileIt.recordIndex(), numRecords - 1);
+        ASSERT_LT(journalFileIt.recordHeader().timestamp(), ts);
+        ASSERT(!journalFileIt.isReverseMode());
+    }
+
+    {
+        // Timestamp less than first record in the file
+        const bsls::Types::Uint64 ts = journalFile.timestampIncrement() / 2;
+        mqbs::JournalFileIterator journalFileIt(
+            &journalFile.mappedFileDescriptor(),
+            journalFile.fileHeader(),
+            false);
+        // Move the iterator to the beginning of the file
+        ASSERT_EQ(journalFileIt.nextRecord(), 1);
+        ASSERT_EQ(m_bmqstoragetool::moveToLowerBound(&journalFileIt, ts), 1);
+        ASSERT_EQ(journalFileIt.recordIndex(), 0);
+        ASSERT_GT(journalFileIt.recordHeader().timestamp(), ts);
+        ASSERT(!journalFileIt.isReverseMode());
+    }
+}
+
+// ============================================================================
+//                                 MAIN PROGRAM
+// ----------------------------------------------------------------------------
+
+int main(int argc, char* argv[])
+{
+    TEST_PROLOG(mwctst::TestHelper::e_DEFAULT);
+
+    switch (_testCase) {
+    case 0:
+    case 1: test1_breathingTest(); break;
+    case 2: test2_searchGuidTest(); break;
+    case 3: test3_searchNonExistingGuidTest(); break;
+    case 4: test4_searchExistingAndNonExistingGuidTest(); break;
+    case 5: test5_searchOutstandingMessagesTest(); break;
+    case 6: test6_searchConfirmedMessagesTest(); break;
+    case 7: test7_searchPartiallyConfirmedMessagesTest(); break;
+    case 8: test8_searchMessagesByQueueKeyTest(); break;
+    case 9: test9_searchMessagesByQueueNameTest(); break;
+    case 10: test10_searchMessagesByQueueNameAndQueueKeyTest(); break;
+    case 11: test11_searchMessagesByTimestamp(); break;
+    case 12: test12_printMessagesDetailsTest(); break;
+    case 13: test13_searchMessagesWithPayloadDumpTest(); break;
+    case 14: test14_summaryTest(); break;
+    case 15: test15_timestampSearchTest(); break;
+    default: {
+        cerr << "WARNING: CASE '" << _testCase << "' NOT FOUND." << endl;
+        s_testStatus = -1;
+    } break;
+    }
+
+    // TODO: consider enable e_CHECK_DEF_GBL_ALLOC
```

Comment:
They did, but I've refactored the tests and `SearchResult...` classes to avoid default allocator usage. Now we can specify `e_CHECK_DEF_GBL_ALLOC` here. Please note, that I also had to modify `mqbs::FileStoreProtocolPrinter::printHeader()` function by adding default parameter `allocator`.

Reply:
Sounds good